bindings/java/src/org/sleuthkit/datamodel/DBAccessManager.java

```diff
@@ -0,0 +1,287 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.logging.Logger;
+import org.sleuthkit.datamodel.SleuthkitCase.CaseDbConnection;
+import static org.sleuthkit.datamodel.SleuthkitCase.closeStatement;
+import org.sleuthkit.datamodel.TskData.DbType;
+
+/**
+ * This class provides modules with access to the case database 
+ * to create custom tables/indexes and to query them.
+ *
+ */
+public final class DBAccessManager {
+
+	private static final Logger logger = Logger.getLogger(DBAccessManager.class.getName());
+
+	private final SleuthkitCase tskDB;
+
+	/**
+	 * Constructor
+	 *
+	 * @param skCase The SleuthkitCase
+	 *
+	 */
+	DBAccessManager(SleuthkitCase skCase) {
+		this.tskDB = skCase;
+	}
+
+	/**
+	 * Creates a table with the specified name and schema.
+	 * 
+	 * @param tableName name of the table to create
+	 * @param tableSchema table schema
+	 * 
+	 * @throws TskCoreException 
+	 */
+	public void createTable(final String tableName, final String tableSchema) throws TskCoreException {
+
+		validateTableName(tableName);
+		validateSQL(tableSchema);
+
+		CaseDbConnection connection = tskDB.getConnection();
+		tskDB.acquireSingleUserCaseReadLock();
+
+		Statement statement = null;
+		String createSQL = "CREATE TABLE IF NOT EXISTS " + tableName + " " + tableSchema;
+		try {
+			statement = connection.createStatement();
+			statement.execute(createSQL);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error creating table " + tableName, ex);
+		} finally {
+			closeStatement(statement);
+			connection.close();
+			tskDB.releaseSingleUserCaseReadLock();
+		}
+
+	}
+
+	/**
+	 * Creates an index on the specified table, on specified column(s).
+	 * 
+	 * @param indexName name of index to create
+	 * @param tableName name of table to create the index on
+	 * @param colsSQL - columns on which to index
+	 * 
+	 * @throws TskCoreException 
+	 */
+	public void createIndex(final String indexName, final String tableName, final String colsSQL) throws TskCoreException {
+
+		validateTableName(indexName);
+		validateSQL(colsSQL);
+
+		CaseDbConnection connection = tskDB.getConnection();
+		tskDB.acquireSingleUserCaseReadLock();
+
+		Statement statement = null;
+		String indexSQL = "CREATE INDEX IF NOT EXISTS " + indexName + " ON " + tableName + " " + colsSQL; // NON-NLS
```

Comment:
Does this fail quietly if the index already exists? Is this the correct syntax for both PostgreSQL and SQLite?

Reply:
Yes, it "does nothing", if the index already exists.  And yes, the syntax covers both DB variants.

Reply:
Good, thanks.